src/screens/SendConfirmationSheet.tsx

```diff
@@ -376,25 +377,33 @@ export const SendConfirmationSheet = () => {
 
   const insufficientEth = isSufficientGas === false && isValidGas;
 
-  const handleSubmit = useCallback(async () => {
-    if (!canSubmit) return;
-    try {
-      setIsAuthorizing(true);
-      if (isENS) {
-        const clearRecords = checkboxes.some(({ checked, id }) => checked && id === 'clear-records');
-        const setAddress = checkboxes.some(({ checked, id }) => checked && id === 'set-address');
-        const transferControl = checkboxes.some(({ checked, id }) => checked && id === 'transfer-control');
-        await callback({
-          ens: { clearRecords, setAddress, transferControl },
-        });
-      } else {
-        await callback();
-      }
-    } catch (e) {
-      logger.sentry('TX submit failed', e);
-      setIsAuthorizing(false);
-    }
-  }, [callback, canSubmit, checkboxes, isENS]);
+  const handleSubmit = useCallback(
+    () =>
+      performanceTracking.getState().executeFn({
+        fn: async () => {
+          if (!canSubmit) return;
+          try {
+            setIsAuthorizing(true);
+            if (isENS) {
+              const clearRecords = checkboxes.some(({ checked, id }) => checked && id === 'clear-records');
+              const setAddress = checkboxes.some(({ checked, id }) => checked && id === 'set-address');
+              const transferControl = checkboxes.some(({ checked, id }) => checked && id === 'transfer-control');
+              callback({
```

Comment:
can we bring back the `awaits` here so that we don't change the flow of handling via the try/catch